src/main/scala/io/scalac/wtf/domain/UserService.scala

```diff
@@ -14,14 +16,15 @@ object UserService {
     def pure[A](a: A): DBIO[A] = DBIO.successful(a)
     def flatMap[A, B](fa: DBIO[A])(f: A => DBIO[B]) = fa.flatMap(f)
   }
-
+  
   def createUser(createdUser: User)(implicit ec: ExecutionContext): DBIO[Xor[NonEmptyList[ValidationError], UserId]] = {
     val resT = for {
       validUser <- XorT[DBIO, NonEmptyList[ValidationError], User](User.validateUser(createdUser.email, createdUser.password))
-      userId    <- XorT.right[DBIO, NonEmptyList[ValidationError], UserId](UserRepository.save(validUser))
+      userId    <- XorT.right[DBIO, NonEmptyList[ValidationError], UserId](UserRepository.save(hashUserPassword(validUser)))
     } yield userId
 
     resT.value
   }
 
+  private def hashUserPassword(user: User): User = user.copy(password = user.password.bcrypt)
```

Comment:
I think we could model this in a better way. I was thinking along the lines:
1. The request to create user is represented as `case class CreateUserRequest(email: String, password: String)`
2. `User.vaidateUser` will return `case class NewUser(email: String, passwordHash: String)`.
3. UserRepository will have `NewUser` in its `save` method.

This way we can only save a hashed password.
I we're doing this we should also change the database field name to `passwordHash`. 
We should also consider using value classes to represent clear text password and a password hash.

@pjazdzewski1990 @margorczynski ?